### DISCUSSION TITLE
Config.AllowAclUrlReservation = false; causes ObjectDisposedException

### DISCUSSION THREAD

**Discussion by c0nnex** [ServiceStack/Discuss#43] [Ideas]
ServiceStack 6.4.0
Selfhosted App derived from AppSelfHostBase with
Config.AllowAclUrlReservation = false 
to enable showing a custom information to the user why app requests UAC now.

Second call of ServiceStack.Start(listenOn) after manually calling HttpListenerBase.AddUrlReservationToAcl(listenOn); throws a ObjectDisposed-Exception on 
ServiceStack.Host.HttpListener.HttpListenerBase.Listener 

Workaround:

```c#
internal void ResetListener() {
            if (Listener != null)
            {
                if (Listener.IsListening)
                    Listener.Stop();
                Listener = null;
            }
        }

```
to enfore creation of a new listener. Very ugly. 
How about providing a Callback action to Host prior to doing the AddUrlReservationToAcl to enable showing of a custom information.
Most users will click cancel if an app requests admin previleges and they do not know why (which is generally what i advise to them as well. Any app requesting admin priv, and you have no idea why? deny it)

**Comment by mythz** [comment]
Feel free to send a PR to [github.com/ServiceStack/ServiceStack](https://github.com/ServiceStack/ServiceStack) with what you need.
